Zend/zend_vm_def.h

```diff
@@ -8935,7 +8935,29 @@ ZEND_VM_COLD_CONST_HANDLER(197, ZEND_MATCH_ERROR, CONST|TMPVARCV, UNUSED)
 
 	SAVE_OPLINE();
 	op = GET_OP1_ZVAL_PTR_UNDEF(BP_VAR_R);
-	zend_throw_exception_ex(zend_ce_unhandled_match_error, 0, "Unhandled match value of type %s", zend_zval_type_name(op));
+
+	// For simpler types where there is a convenient stringified version, include that
+	// in the error message. Otherwise just show its type.
+	switch (Z_TYPE_P(op)) {
+	case IS_FALSE:
+		zend_throw_exception_ex(zend_ce_unhandled_match_error, 0, "Unhandled match value: false");
+		break;
+	case IS_TRUE:
+		zend_throw_exception_ex(zend_ce_unhandled_match_error, 0, "Unhandled match value: true");
+		break;
+	case IS_LONG:
+	case IS_DOUBLE:
+	case IS_STRING: {
```

Comment:
I don't know if we have a convention in PHP or not, but I have not seen `case` statements in code blocks. The style is not even consistent in this PR, so please remove the code fences from this `case`

Reply:
Again, this was retained from the example linked above.  Since it's declaring a new variable, the `{}` creates a local scope so that the variable doesn't persist past the block.  In this case I don't think it matters, but that's what the `php_debug_zval_dump()` function does.

Reply:
Ah ok, that makes sense then.

Reply:
C89 doesn't allow to mix declarations and statements. As of PHP 8.0.0, we're officially requiring C99 which does allow that; still, the block makes sense in this case.